File: libs/python/edgecommons/platform/transport.py
"""
The messaging *transport* — the secondary runtime axis (DESIGN-core sec 2).

Defaults from the resolved :class:`~edgecommons.platform.platform.Platform`
(GREENGRASS -> IPC, HOST -> MQTT) and is independently overridable, but
constrained: :attr:`IPC` is valid only on
:attr:`~edgecommons.platform.platform.Platform.GREENGRASS` (the Nucleus provides
the IPC socket). See :func:`~edgecommons.platform.resolver.validate`.
"""

from enum import Enum


class Transport(str, Enum):
    """Messaging transport (str-valued so it compares to the raw uppercased CLI token)."""

    #: Greengrass Nucleus IPC (domain socket). Requires platform GREENGRASS.
    IPC = "IPC"
    #: Dual-MQTT (local broker + AWS IoT Core). The off-Nucleus transport.
    MQTT = "MQTT"

File: libs/java/src/main/java/com/mbreissi/edgecommons/platform/Transport.java
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */
package com.mbreissi.edgecommons.platform;

/**
 * The messaging <em>transport</em> — the secondary runtime axis (DESIGN-core §2). Defaults from the
 * resolved {@link Platform} (GREENGRASS&rarr;IPC, HOST&rarr;MQTT) and is independently overridable,
 * but constrained: {@link #IPC} is valid only on {@link Platform#GREENGRASS} (the Nucleus provides
 * the IPC socket). See {@link PlatformResolver#validate}.
 */
public enum Transport {
    /** Greengrass Nucleus IPC (domain socket). Requires {@link Platform#GREENGRASS}. */
    IPC,
    /** Dual-MQTT (local broker + AWS IoT Core). The off-Nucleus transport. */
    MQTT
}
